File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.py
class Solution(object):
    def subarraysDivByK(self, nums, k):
        freq = [0] * k
        freq[0] = 1
        sum = 0
        ans = 0

        for num in nums:
            sum = (sum + num % k + k) % k
            ans += freq[sum]
            freq[sum] += 1

        return ans

File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& nums, int k) {
        vector<int> freq(k, 0);
        freq[0] = 1;
        int sum = 0;
        int ans = 0;

        for(int i=0; i<nums.size(); i++) {
            sum = (sum + nums[i] % k + k) % k;
            ans += freq[sum];
            freq[sum]++;
        }

        return ans;   
    }
};